c++/src/H5Attribute.cpp

```diff
@@ -607,7 +607,8 @@ Attribute::~Attribute()
 
 //--------------------------------------------------------------------------
 // Function:    Copy assignment operator
-Attribute & Attribute::operator= (const Attribute &other)
+Attribute &
+Attribute::operator=(const Attribute &other)
 {
     if (&other != this) {
     }
```

Comment:
Done. This is the crux of the WIP. I'm not sure what to actually put into these copy assignment operators...

Reply:
I would say that a copy in this context is the equivalent of (re-)opening the thing in the same way, so you'd want a new, different ID. Unfortunately, we don't have a lightweight way of duplicating an ID, though that could be added pretty easily. There are calls that do what the library would do (e.g., H5Iobject_verify() and H5Iregister()), but we don't allow using those with library ID types (just user-defined ID types).

Reply:
In the HDF5 C++ API, the operator= uses the same ID as the original because if a different ID is created, it won't have the same characteristics as the original ID, specifically, if the rhs represents a named datatype, "this" would still be a transient datatype.  At least, that was my understanding at the time.  However, I just realized PropList didn't follow that concept but I don't remember why...